Update driver to work with cheffish-5.0 and provisioning-2.1.1

jjlimepoint opened the pull request:
MergedConfig changed to be a mash, whcih stringifies things... take the same approch the aws driver did and convert them back inside

bender-the-greatest committed:
If specified, wait for ipv4 address.

bender-the-greatest committed:
Updated README.md

rlanore committed:
Add proxy support

jjlimepoint committed:
ensure machine_options and configs are symbolised, since this is required for cheffish-5.0

jjasghar commented:
Hi! If you're still interested in getting this merged, please close and reopen this
PR against the new maintainers at https://github.com/chef-partners/chef-provisioning-vsphere.

We are in the process of reinvigorating this project and look forward to many collaborations
in the future!

Thanks for your time and help!

Oh! We also are on the [VMware{Code}][slack] slack team in #chef if you'd ever like to come by and
say hi!

[slack]: https://code.vmware.com/web/code/join

Commit:
Updated README and gemfile

- First pass at cleaning up and updating

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
Rubocop wonderness

- Standarizing on the new chef rubocop wonderfulness

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
Converted options to symbols

In order for the ssh connection to work, symbols must be there, not strings

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
Removed debug gemfile

Removed debugging gems from Gemfile

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
Added readme updates and travis

Starting travis for style checks

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
Added github changelog

Added github changelog generator and rake tasks

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
removed pry

Rubocop and pry

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
Merge pull request #2 from chef-partners/cleanup

Cleanup for new release

Commit:
Added links to the README

Added  links to the readme.

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
Merge pull request #3 from chef-partners/cleanup

Added links to the README

Commit:
v1.0.0 release

Starting from 1.0.0 now. This gem has the MVP of what needs to work :)

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
freeze those mutable objects

rubocop was yelling at me.

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
Merge pull request #4 from chef-partners/1.0.0

v1.0.0 release

Commit:
Updated some depenant gems

- Updated rbvmomi
- Updated chef-provisioning
- Pinned to version chef 12
- pinned version of cheffish
- added debug gems

The chef 12 and cheffish pinning is due to the cheffish 5 release and
chef-provisioning lagging behind. As soon as it's released and fixed we
will retest with Chef 13 and Cheffish 5.

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
document reqd fields for sysprep

Commit:
Merge pull request #11 from nealbrown/master

document reqd fields for sysprep

Commit:
Wordified readme

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
Merge pull request #10 from chef-partners/Gemfile_fixes

Updated some depenant gems

Commit:
Merge pull request #8 from chef-partners/metalseargolid-feature/ipv4_bootstrap

Metalseargolid feature/ipv4 bootstrap

Commit:
Added case for HTTP_PROXY

Added changes for https and http proxy

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
Cleaner way to do the parsing

proxy parsing can be challanging.

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
Merge branch 'master' into rlanore-master

Commit:
Merge pull request #7 from chef-partners/rlanore-master

Add proxy support

Commit:
Rubocop and specs

- rubocop'd
- added specs to travis

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
v1.1.0

Updated version and Changelog

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
rubocop'd

Need to make that rubocop happy

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
Merge pull request #12 from chef-partners/1.1.1

rubocop'd

Commit:
v1.1.1

- Rubocop and changelog

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
rake unit, not rake spec

Opps.

Signed-off-by: JJ Asghar <jj@chef.io>

juanxhos committed:
Tested enhancement check host and port available

Commit:
Tested enhancement

juanxhos committed:
Setting attribute first nic ip of the vm, it get lost on ip_to_bootstrap sdk invoke (SDK looks like bug)

juanxhos committed:
Mandatory valid ipv4

juanxhos committed:
Saved valid ipv4 attribute class

juanxhos committed:
Doh lint

juanxhos committed:
Valid saved ipv4 static

juanxhos committed:
Doh Lint grrr

juanxhos committed:
Change socket tcp from socket connect for cases of static ip and dhcp ip

juanxhos committed:
Final tweaks

juanxhos committed:
Linting

Commit:
Merge pull request #14 from jcalonsoh/wait_for_ipv4_enhancement

Wait for ipv4 enhancement

Commit:
Revert "Wait for ipv4 enhancement"

Commit:
Merge pull request #16 from chef-partners/revert-14-wait_for_ipv4_enhancement

Revert "Wait for ipv4 enhancement"

michaeltlombardi committed:
Update README to cover multi-nodes in kitchen.yml

This patch updates the README to clarify the ability of a user to create and provision multiple nodes via test-kitchen and this driver. It shows the users how to overwrite driver settings in both the platform and suite sections. Finally, it gives the helpful tip for users to call `kitchen diagnose` to verify their settings prior to `kitchen create` execution.

Commit:
Merge pull request #18 from michaeltlombardi/patch-1

Update README to cover multi-nodes in kitchen.yml

tuccimon committed:
corrected winrm port assignment so it does not always equal 5986

Commit:
Merge pull request #19 from tuccimon/port-correction

corrected winrm port assignment so it does not always equal 5986

Commit:
1.1.2 release

- Bug release
- Updated readme

Signed-off-by: JJ Asghar <jj@chef.io>

Commit:
Merge pull request #20 from chef-partners/1.1.2

1.1.2 release

jjlimepoint committed:
allow chef13, since it works now

Signed-off-by: Jaymz Julian <jjulian@limepoint.com>

jjlimepoint commented:
closing since wrong tree :)
